internal/storage/fs/snapshot_test.go

```diff
@@ -0,0 +1,987 @@
+				NamespaceKey: "staging",
+				Constraints: []*flipt.Constraint{
+					{
+						SegmentKey:   "segment2",
+						Property:     "foo",
+						Operator:     "eq",
+						Value:        "baz",
+						NamespaceKey: "staging",
+					},
+					{
+						SegmentKey:   "segment2",
+						Property:     "fizz",
+						Operator:     "neq",
+						Value:        "buzz",
+						NamespaceKey: "staging",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			sgmt, err := fis.store.GetSegment(context.TODO(), tc.namespace, tc.segmentKey)
+			assert.NoError(t, err)
+
+			assert.Equal(t, tc.segment.Key, sgmt.Key)
+			assert.Equal(t, tc.segment.Name, sgmt.Name)
+			assert.Equal(t, tc.segment.NamespaceKey, sgmt.NamespaceKey)
+
+			for i := 0; i < len(tc.segment.Constraints); i++ {
+				c := tc.segment.Constraints[i]
+				fc := sgmt.Constraints[i]
+				assert.Equal(t, c.SegmentKey, fc.SegmentKey)
+				assert.Equal(t, c.Property, fc.Property)
+				assert.Equal(t, c.Operator, fc.Operator)
+				assert.Equal(t, c.Value, fc.Value)
+				assert.Equal(t, c.NamespaceKey, fc.NamespaceKey)
+			}
+		})
+	}
+}
+
+func (fis *FSWithoutIndexSuite) TestCountSegment() {
+	t := fis.T()
+
+	segmentCount, err := fis.store.CountSegments(context.TODO(), "production")
+	assert.NoError(t, err)
+
+	assert.Equal(t, 12, int(segmentCount))
+
+	segmentCount, err = fis.store.CountSegments(context.TODO(), "sandbox")
+	assert.NoError(t, err)
+	assert.Equal(t, 12, int(segmentCount))
+
+	segmentCount, err = fis.store.CountSegments(context.TODO(), "staging")
+	assert.NoError(t, err)
+	assert.Equal(t, 12, int(segmentCount))
+}
+
+func (fis *FSWithoutIndexSuite) TestListSegments() {
+	t := fis.T()
+
+	testCases := []struct {
+		name      string
+		namespace string
+		pageToken string
+	}{
+		{
+			name:      "Production",
+			namespace: "production",
+		},
+		{
+			name:      "Sandbox",
+			namespace: "sandbox",
+		},
+		{
+			name:      "Staging",
+			namespace: "staging",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			flags, err := fis.store.ListSegments(context.TODO(), tc.namespace, storage.WithLimit(5), storage.WithPageToken("0"))
+			assert.NoError(t, err)
+
+			assert.Len(t, flags.Results, 5)
+			assert.Equal(t, "5", flags.NextPageToken)
+		})
+	}
+}
+
+func (fis *FSWithoutIndexSuite) TestGetEvaluationRules() {
+	t := fis.T()
+
+	testCases := []struct {
+		name      string
+		namespace string
+		flagKey   string
+	}{
+		{
+			name:      "Production",
+			namespace: "production",
+			flagKey:   "prod-flag",
+		},
+		{
+			name:      "Sandbox",
+			namespace: "sandbox",
+			flagKey:   "sandbox-flag",
+		},
+		{
+			name:      "Sandbox",
+			namespace: "staging",
+			flagKey:   "staging-flag",
+		},
+	}
+
+	for _, tc := range testCases {
+		dist, err := fis.store.GetEvaluationRules(context.TODO(), tc.namespace, tc.flagKey)
+		assert.NoError(t, err)
+
+		assert.Len(t, dist, 1)
+	}
```

Comment:
Could we check some more specific details here.
This is what `Evaluate` uses to make its decisions. So would be good to understand what rules it is going to attempt.